compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -577,6 +579,23 @@ setMatmulVectorDistributionConfig(mlir::FunctionOpInterface entryPoint,
         deduceMMASchedule(problem, intrinsics, seeds, sharedMemoryLimitInBytes,
                           /*canUpcastAcc=*/true);
   }
+
+  // Only batch_matmul is supported in the LLVMGPUPadAndVectorDistribute
+  // pipeline.
+  if (!schedule && !contractionDims->batch.empty()) {
+    pipeline = IREE::Codegen::DispatchLoweringPassPipeline::
+        LLVMGPUPadAndVectorDistribute;
+    bool mustBeAligned = false;
```

Comment:
If I understand this strategy correctly, it will break if there are fused producers. Can you add a check here for that? Can use this helper: https://github.com/iree-org/iree/blob/3ca0a49b424c7d9918a2e73d0c19c308e7d8e6db/compiler/src/iree/compiler/Codegen/Utils/Utils.cpp#L1115

Reply:
The strategy works for `fill + batch_matmul`. Won't it disable the support for the case?

Reply:
That helper only looks at inputs, it ignores inits so it should be fine.